### PR TITLE
CRM-18542 - Make Date Display Format change easier and faster

### PR DESCRIPTION
**Pull request by rajbdilip**
https://issues.civicrm.org/jira/browse/CRM-18542
Replaced textboxes with dropdowns with common date and/or time formats
on Administer > Localization > Date Formats page to make the date format change
process much easier and faster.

### CODE REVIEW

**Comment by monishdeb** [comment]
You got two indentation warning, check the related test build failure https://test.civicrm.org/job/CiviCRM-Core-PR/9321/checkstyleResult/new/

**Comment by rajbdilip** [comment]
@monishdeb Fixed.

**Comment by monishdeb** [comment]
unfortunately it fails again :/ https://test.civicrm.org/job/CiviCRM-Core-PR/9322/checkstyleResult/new/

**Comment by monishdeb** [comment]
Also you can squash multiple commit into one by 
1) git rebase --interactive
2) vi editor will open 
    edit the latest commits (2 in your case) from pick to squash like 
- pick commit1 
- squash commit2
- squash commit3 
3) Save and remove the commit message 
4) then 'git push -f origin CRM-18542

**Comment by rajbdilip** [comment]
@monishdeb Done. :D

**Comment by monishdeb** [comment]
Cool :)

**Comment by rajbdilip** [comment]
@monishdeb Still?

**Comment by rajbdilip** [comment]
Jenkins, retest this please.

**Comment by monishdeb** [comment]
Since you have changed the default date formats on localisation.setting.php it has affected 6 unit tests which expects date in '%B %E%f, %Y' format but given '%m/%d/%Y' as those dates in failing unit tests are asserted on hardcoded date text. So in my opinion its better to ONLY retain previous default rather than making adjustment in failing unit tests and others also may like the previous default format.

**Comment by rajbdilip** [comment]
@monishdeb Finally! :D

**Comment by rajbdilip** [comment]
@monishdeb Could you please look into it and merge it?

**Comment by JoeMurray** [comment]
Congrats, @rajbdilip ! Thanks @monishdeb

File: CRM/Core/SelectValues.php [hunk 1]
```diff
@@ -779,6 +779,96 @@ public static function datePluginToPHPFormats() {
...
+  }
+
+  /**
+   * CiviCRM supported time display formats.
+   *
+   * @return array
+   */
+  public static function getDatePluginDisplayFormatsTime() {
+    $dateDisplayFormats = array(
+      "%l:%M %P" => ts('hh:mm A( 7:30 PM)'),
```
Comment (colemanw) [inline CRM/Core/SelectValues.php]: space should be after the `A` instead of after the `(`

File: CRM/Core/SelectValues.php [hunk 2]
```diff
@@ -779,6 +779,96 @@ public static function datePluginToPHPFormats() {
...
+      "%H:%M" => ts('HH:mm(19:30)'),
```
Comment (colemanw) [inline CRM/Core/SelectValues.php]: Need a space before the `(`

**Comment by rajbdilip** [comment]
@colemanw Corrected the lines you pointed out. Also updated the PR to replace dropdown for `Financial Batch Date` field which was [recently added](https://github.com/civicrm/civicrm-core/pull/8441).

File: settings/Localization.setting.php [hunk 3]
```diff
@@ -333,13 +340,13 @@
...
-    'html_attributes' => array(
-      'size' => '12',
-      'maxlength' => '60',
+    'quick_form_type' => 'Select',
+    'html_type' => 'Select',
+    'pseudoconstant' => array(
+      'callback' => 'CRM_Core_SelectValues::getDatePluginDisplayFormatsFinancialBatch',
     ),
     'default' => '%m/%d/%Y',
+    'add' => '4.3',
```
Comment (yashodha) [inline settings/Localization.setting.php]: Shouldn't this be 4.7?

Comment (rajbdilip) [inline settings/Localization.setting.php]: @yashodha Fixed. :)

**Comment by colemanw** [comment]
Closing per discussion on [jira ticket](https://issues.civicrm.org/jira/browse/CRM-18542).

**Pull request closed without merging**